src/lib.rs

```diff
@@ -55,13 +52,41 @@ pub struct Backend {
     extras: HashMap<String, String>,
 }
 
-struct ProxyState {
+struct ProxyConfig {
     config: Config,
     client: Client,
 }
+impl ProxyConfig {
+    fn new(config: Config, client: Client) -> ProxyConfig {
+        ProxyConfig { config, client }
+    }
+}
+
+#[derive(Debug)]
+pub struct BackendState {
+    rr_count: isize,
+}
+struct ProxyState {
+    backends: HashMap<String, Option<BackendState>>,
+}
+
 impl ProxyState {
-    fn new(config: Config, client: Client) -> ProxyState {
-        ProxyState { config, client }
+    fn new(config: &Config) -> ProxyState {
+        let mut backends: HashMap<String, Option<BackendState>> = HashMap::new();
+
+        for backend_config in &config.backends {
+            if backend_config.backend_type == Some("loadbalanced".to_string())
```

Comment:
You should be able to do: `backend_config.backend_type.as_deref() == Some("loadbalanced")`